tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/ClientUpdate/ClientUpdateLanguageServiceResolver.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Microsoft.Extensions.Logging;
+using Azure.Sdk.Tools.Cli.Services; // for ILanguageSpecificCheckResolver
+
+namespace Azure.Sdk.Tools.Cli.Services.ClientUpdate;
+
+/// <summary>
+/// Resolves an <see cref="IClientUpdateLanguageService"/> implementation based on detected package language.
+/// Reuses existing language detection logic via ILanguageSpecificCheckResolver + individual service SupportedLanguage values.
+/// </summary>
+public interface IClientUpdateLanguageServiceResolver
+{
+    /// <summary>
+    /// Attempts to resolve an <see cref="IClientUpdateLanguageService"/> for the supplied generated package path.
+    /// </summary>
+    /// <param name="packagePath">Root directory of a generated SDK package whose language needs to be inferred. May be <c>null</c> or non-existent.</param>
+    /// <param name="ct">Cancellation token.</param>
+    /// <returns>
+    /// Matching language service when the language can be determined and a service is registered;
+    /// otherwise the first registered service (best-effort fallback), or <c>null</c> if none are registered.
+    /// </returns>
+    Task<IClientUpdateLanguageService?> ResolveAsync(string? packagePath, CancellationToken ct = default);
+}
+
+/// <summary>
+/// Default implementation of <see cref="IClientUpdateLanguageServiceResolver"/> which uses the shared
+/// <see cref="ILanguageSpecificCheckResolver"/> to detect a package's language, then matches the language
+/// against registered <see cref="IClientUpdateLanguageService.SupportedLanguage"/> values. Provides defensive
```

Comment:
Can you give an example of a reasonable fallback we could use here? I'm struggling with it - each updater is going to be language specific. I can't see using them across implementations.

Or is this temporary until we get a specific updater per language? (ie, eventually it'd be an error if nothing comes back).